### ISSUE TITLE
Remove unnecessary @function tags

### ISSUE THREAD

**Issue by pjcozzi** [AnalyticalGraphicsInc/cesium#3341]
For example, `@function` is needed here:
```javascript
    /**
     * Given the desired degree, returns the number of data points required for interpolation.
     * @function
     *
     * @param {Number} degree The desired degree of interpolation.
     * @returns {Number} The number of required data points needed for the desired degree of interpolation.
     */
    InterpolationAlgorithm.getRequiredDataPoints = DeveloperError.throwInstantiationError;
```
AFAIK it is not needed here:
```javascript
    /**
     * Stores the provided instance into the provided array.
     * @function
     *
     * @param {BoxGeometry} value The value to pack.
     * @param {Number[]} array The array to pack into.
     * @param {Number} [startingIndex=0] The index into the array at which to start packing the elements.
     */
    BoxGeometry.pack = function(value, array, startingIndex) {
        // ...
```

**Comment by mramato** [comment]
You are correct.  The rule is that `@function` is only needed when JSDoc can't infer that it is a function (such as your first example). Link: http://usejsdoc.org/tags-function.html

Having `@function` where it isn't needed is also completely harmless (as long as the think is a function), but I guess we should be consistent in when we use it.

**Comment by AmayAgrawal** [comment]
I am looking over this issue,but I didn't get exactly when to use @function and when not.Can anyone please explain it more briefly??

**Comment by pjcozzi** [comment]
Thanks for looking into this, @AmayAgrawal!

`@function` is only needed when the JavaScript function declaration is not there.  In Cesium, this is needed when we assign one function to another, e.g., this needs `@function`:
```javascript
    /**
     * Given the desired degree, returns the number of data points required for interpolation.
     * @function
     *
     * @param {Number} degree The desired degree of interpolation.
     * @returns {Number} The number of required data points needed for the desired degree of interpolation.
     */
    InterpolationAlgorithm.getRequiredDataPoints = DeveloperError.throwInstantiationError;
```

If you search for `@function` in all of the Cesium *.js files, it should be pretty easy to identify the ones that are not needed and remove them.

Then, just double check your work by building and viewing the documentation.

**Comment by pjcozzi** [comment]
Fixed in #3375.
